src/suffer.cpp

```diff
@@ -1473,7 +1473,12 @@ void Character::suffer()
     }
 
     for( size_t i = 0; i < my_bionics->size(); i++ ) {
-        if( ( *my_bionics )[i].powered ) {
+        bionic &bio = ( *my_bionics )[i];
+        if( calendar::once_every( 1_hours ) && !bio.id->fuel_opts.empty() &&
+            bio.has_flag( "AUTO_START_ON" ) && power_level <= 0_mJ ) {
+            g->u.activate_bionic( i );
```

Comment:
Why are you activating the bionic on `g->u`? What if the player character doesn't have a bionic with index `i`?

Reply:
Activate is in player and apprently I can't call it from Character, also since the flag can only be used by the player I thought that it shouldn't cause any issue

Reply:
> Activate is in player and apprently I can't call it from Character,

Yeah, the class hierarchy is totally screwed at that point. Anyway, it's just not future proof as `Character::suffer` could get called on an NPC object at some time in the future.

>  also since the flag can only be used by the player

Yes, it will work for now.

----

I see this kind of logic from time to time (checking the implicit `this` object, than calling functions of `g->u`) and I want to make sure this is intentional as `this` and `g->u` may not be the same object all the time.